src/pages/faq/faq.mdx

```diff
@@ -0,0 +1,133 @@
+For this purpose you can use the `onChange` prop, which returns the form data and possible validation errors provided by AJV.
+To get you started, we built two simple examples for you. 
+
+The first example showcases a class component:
+```
+import React from 'react';
+import { JsonForms } from '@jsonforms/react';
+import { person } from '@jsonforms/examples';
+import { materialRenderers } from '@jsonforms/material-renderers';
+
+const schema = person.schema;
+const uischema = person.uischema;
+const data = person.data;
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data
+    };
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <JsonForms
+          schema={schema}
+          uischema={uischema}
+          data={data}
+          renderers={materialRenderers}
+          onChange={({ errors, data }) => this.setState({data})}
+        />
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+The second example uses the functional component approach. Therefore we use the `useState` hook, which provides the data (first parameter) and a function to update the data (second parameter).
+```
+import React, { useState } from 'react';
+import { JsonForms } from '@jsonforms/react';
+import { person } from '@jsonforms/examples';
+import { materialRenderers } from '@jsonforms/material-renderers';
+
+const schema = person.schema;
+const uischema = person.uischema;
+const data = person.data;
+
+function App() {
+  const [stateData, setData] = useState(data);
+
+  return (
+    <div className="App">
+      <JsonForms
+        schema={schema}
+        uischema={uischema}
+        data={stateData}
+        renderers={materialRenderers}
+        onChange={({ errors, data }) => setData(data)}
+      />
+    </div>
+  );
+}
+
+export default App;
+```
+
+
+## How can I integrate a custom renderer?
+
+If you want to change the way a field is rendered (for example showing stars for a rating instead of a number), you can create your own renderer. 
+This requires 3 things:
+
+1. Create a custom renderer
+2. Create a tester (which tells the system for which field this renderer should be used)
+3. Register the custom renderer and the tester
+
+Within this example, we register the custom renderer directly on the JSON Forms component:
+```
+import { rankWith, scopeEndsWith } from '@jsonforms/core';
+import { Rating } from './Rating';
+
+const RatingControl = ({ data, handleChange, path }) => (
+  <Rating
+    value={data}
+    onClick={ev => handleChange(path, Number(ev.value))}
+  />
+);
+
+<JsonForms
+  schema={schema}
+  uischema={uischema}
+  data={stateData}
+  renderers={[
+    ...materialRenderers,
+    //register custom renderer
+    { tester: rankWith(3,scopeEndsWith('rating')), renderer: RatingControl }
+  ]}
+/>
+```
+
+In contrast to the first example, the following code uses the redux store to register the custom renderer:
```

Comment:
When using the redux integration you'll need to use the `registerRenderer` action: (relative link to https://jsonforms.io/docs/available-actions#registerrenderertester-rankedtester-renderer-any)